### PR TITLE
TP-26791 Python 3 upgrade: py-wkhtmltox upgrade

### PR DESCRIPTION
**Pull request by michalplichta-sc**
Python 3 upgrade: py-wkhtmltox upgrade
https://scurri.tpondemand.com/entity/26791-python-3-upgrade-py-wkhtmltox-upgrade

As a Scurri developer, I would like to upgrade the py-wkhtmltox library to support Python 2.7 and Python 3.4+.

The library can be found here:
https://github.com/Scurri/py-wkhtmltox

Tests using `tox` must be implemented to ensure that the upgraded library is supported by both, Python 2.7 and 3.4+.

We must also bump the version so we can later update Scurri requirements.

**Acceptance Criteria**
* The library is upgraded
* Proper tests are implemented
* The version is bumped

**Goal**
* Make the library Python 2 and 3 compatible

**Checks**
 - [x] Acceptance Criteria Satisfied
 - [x] Code changes are covered by test suite
 - [x] Pull request is properly formatted

### CODE REVIEW

File: wkhtmltox.pyx [hunk 1]
```diff
@@ -69,10 +69,10 @@ cdef class _Pdf:
...
         pass
-        #wkhtmltopdf_deinit();
+        #wkhtmltopdf_deinit()
     
     def version(self):
-        return wkhtmltopdf_version();
+        return wkhtmltopdf_version()
     
     def set_global_setting(self, char *name, char *value):
         return wkhtmltopdf_set_global_setting(self._c_global_settings, name, value)
```
Comment (brian-barba-hernandez) [inline wkhtmltox.pyx]: There's `iteritems` down

File: setup.cfg [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+version = 0.3
+author = Matt Reiferson
+author_email = mreiferson@gmail.com
+description = Python bindings for libwkhtmltox
+url = http://github.com/mreiferson/py-wkhtmltox
+
+[options]
+include_package_data = True
+packages = find:
+python_requires = >=2.7, !=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*, <3.5
```
Comment (brian-barba-hernandez) [inline setup.cfg]: Do not restrict to `<3.5`, we don't want to update the library if it doesn't require changes for higher versions

**Review by brian-barba-hernandez — CHANGES_REQUESTED** [review]
As there's a leftover for `iteritems`, this shows that we haven't covered other places, could you improve the coverage and add a report about how much we're covering?

File: setup.cfg (hunk 2 above)
Comment (michalplichta-sc) [reply setup.cfg]: done

File: wkhtmltox.pyx (hunk 1 above)
Comment (michalplichta-sc) [reply wkhtmltox.pyx]: good catch, missed `;` and `iteritems`

File: tests/test_wkhtmltox.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    img.convert()
+
+    assert os.path.isfile(test_file)
+    with open('tests/simple.jpg', 'rb') as org, open(test_file, 'rb') as new:
+        assert assert_images_are_identical(
+            'test_image', org.read(), new.read(), 'jpg')
+    try:
+        os.remove(test_file)
+    except OSError:
+        pass
```
Comment (brian-barba-hernandez) [inline tests/test_wkhtmltox.py]: if the assertion fails, the test file remains and it's not deleted.

File: tests/test_wkhtmltox.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    with open('tests/simple.pdf', 'rb') as org, open(test_file, 'rb') as new:
+        assert_pdf_objects_are_identical(
+            'test_pdf',
+            PDF.from_bytes(org.read()),
+            PDF.from_bytes(new.read())
+        )
+    try:
+        os.remove(test_file)
+    except OSError:
+        pass
```
Comment (brian-barba-hernandez) [inline tests/test_wkhtmltox.py]: Ditto

File: tox.ini [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    Cython==0.29.24
+    Pillow==5.4.1
+    PyPDF2==1.25.1
+
+[testenv:py34]
+deps =
+    pytest
+    Cython==0.29.24
+    Pillow==5.4.1
+    PyPDF2==1.25.1
```
Comment (brian-barba-hernandez) [inline tox.ini]: AFAIR, if both environments have the exact same requirements you can just specify them in the `[testenv]` section.

**Review by brian-barba-hernandez — CHANGES_REQUESTED** [review]

File: tests/test_wkhtmltox.py (hunk 3 above)
Comment (michalplichta-sc) [reply tests/test_wkhtmltox.py]: :facepalm:

File: tox.ini (hunk 5 above)
Comment (michalplichta-sc) [reply tox.ini]: good

**Review by brian-barba-hernandez — APPROVED** [review]
Even tho the PDF sample looks ugly, I can confirm that the same HTML produces the same ugly PDF in Scurri docker container.

**Review by dmitrynedbaylo-sc — CHANGES_REQUESTED** [review]
Split PR please: first one regression tests and if possible tox setup (without python3); second one Python 3 upgrade related changes.

**Review by brian-barba-hernandez — APPROVED** [review]

**Review by dmitrynedbaylo-sc — APPROVED** [review]

**Pull request merged**